kpet/data.py

```diff
@@ -451,6 +464,66 @@ def convert_tree_arches(self):
 
             self.trees[name]["arches"] = list(tree_arches)
 
+    def convert_sets(self):
+        """
+        Convert each suite / cases sets from a list of regexes
+        to a list of set names matching those regexes
+
+        Raises:
+            A schema.Invalid exception when finding an invalid regex
+        """
+
+        error = ("One of '{0}: set:' regexes\n" +
+                 "doesn't match any of the available sets\n" +
+                 "The regex: '{1}'\n" +
+                 "The avaliable sets: " + str(list(self.sets.keys())))
+
+        # TODO: Remove after conversion
+        # This function is to convert from the old format (where sets were a
+        # part of the pattern)
+        def convert_pattern(pattern):
+            sets = set(self.sets.keys())
+            matches = set()
+            for set_name in sets:
+                target = Target(trees=Target.ANY,
+                                arches=Target.ANY,
+                                components=Target.ANY,
+                                sets=set([set_name]),
+                                sources=Target.ANY)
+                if pattern.matches(target):
+                    matches.add(set_name)
+            if matches == sets:
+                return set()
```

Comment:
I think it's better to return `None` here instead. Then we'll be able to treat set specification absent from patterns the same way we treat absent `sets` property, and make our code more uniform. Oh, and an empty set would be a valid case too.

Reply:
This is the function to convert patterns into a set of sets, this return happens if the pattern doesn't specify that it wants a specific set, thus it will match every sets (matches will be all sets) and in that case we don't want to add any sets to whichever sets are already specified by the 'sets' field of the suite/case. Since I only use this in a scenario where there is `set |= convert_pattern(foo)` returning a None would only add another check.
Not sure if we'd ever want to differenciate between a pattern that doesn't care about sets or one that doesn't match any(because of a typo or smth) and if there is a check for this already because if not I think that'd be the only reason to change this